src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -71,6 +119,107 @@ public void periodic(){
         else {
             driveHardware.arcadeDrive(arcadeForward, arcadeRotation);
         }
-    } 
 
-}
+        for (int i = 0; i < 4; i++) {
+            //Get encoder value
+            Rotation2d moduleRotation = new Rotation2d(driveModules[i].getRotationPosition());
+
+            //Optimize each module state
+            SwerveModuleState optimizedState = SwerveModuleState.optimize(goalModuleStates[i], moduleRotation);
+            driveModules[i].moduleControl(optimizedState, moduleRotation);
```

Comment:
Does moduleControl need moduleRotation as an input? Doesn't it already have that?

Reply:
fixed